### ISSUE TITLE
haxe.io.Input.readLine() and readUntil() screw up string encoding

### ISSUE THREAD

**Issue by sebthom** [HaxeFoundation/haxe#5429]
In the following code the second line evaluates to `false` for the targets C#, Java and Python.

``` haxe
class Test {
    static function main() {
        trace("Äpfel" == haxe.io.Bytes.ofString("Äpfel").toString());
        trace("Äpfel" == new haxe.io.StringInput("Äpfel").readLine());
    }
}
```

The reason is that haxe.io.Input adds each byte read from the input stream using addChar() to StringBuf. Which works e.g. on Neko where the StringBuf is implemented as a byte buffer but fails on other platforms where then each byte would be treated as a separate character ignoring the actual encoding.

Using BytesBuffer instead of StringBuf resolves the issue.

``` haxe
    /**
        Read a string until a character code specified by `end` is occurred.

        The final character is not included in the resulting string.
    **/
    public function readUntil( end : Int ) : String {
        var buf = new BytesBuffer();
        var last : Int;
        while( (last = readByte()) != end )
            buf.addByte( last );
        return buf.getBytes().toString();
    }

    /**
        Read a line of text separated by CR and/or LF bytes.

        The CR/LF characters are not included in the resulting string.
    **/
    public function readLine() : String {
        var buf = new BytesBuffer();
        var last : Int;
        var s;
        try {
            while( (last = readByte()) != 10 )
                buf.addByte( last );
            s = buf.getBytes().toString();
            if( s.charCodeAt(s.length-1) == 13 ) s = s.substr(0,-1);
        } catch( e : Eof ) {
            s = buf.getBytes().toString();
            if( s.length == 0 )
                #if neko neko.Lib.rethrow #else throw #end (e);
        }
        return s;
    }
```

**Comment by nadako** [comment]
The second line returns `false` for me on node.js as well.

**Comment by ncannasse** [comment]
I think that goes with #3072

**Comment by sebthom** [comment]
Is fixed in Haxe 3.4.0.
